core/client/src/main/java/alluxio/client/block/BlockWorkerClient.java

```diff
@@ -219,6 +219,16 @@ private synchronized void connectOperation() throws IOException {
         LOG.error(e.getMessage(), e);
         return;
       }
+
+      // Send a heartbeat to the worker to register the new session id
+      try {
+        mClient.sessionHeartbeat(mSessionId, mClientMetrics.getHeartbeatData());
+      } catch (Exception e) {
+        LOG.error(e.getMessage(), e);
+        // Directly close the transport instead of disconnecting because we are not connected yet
```

Comment:
oh, I thought the comment meant disconnecting the transport, and that the transport wasn't connected yet. I understand now